monix-reactive/shared/src/test/scala/monix/reactive/consumers/ForeachAsyncConsumerSuite.scala

```diff
@@ -101,4 +102,21 @@ object ForeachAsyncConsumerSuite extends TestSuite[TestScheduler] {
     s.tick()
     assert(cancelled)
   }
+
+  test("should suspend effects encountered during the stream") { implicit s =>
+    val exceptionMessage: String = "Boom!"
+    val f: Future[Unit] = {
+      Observable(1)
+        .consumeWith(
+          Consumer.foreachTask(_ =>
+            Task.raiseError(new RuntimeException(exceptionMessage))
+          )
+        ).runToFuture
+    }
+
+    s.tick()
+    assert(f.value.get.isFailure)
+    assertEquals(f.value.get.failed.get.getMessage, exceptionMessage)
```

Comment:
We usually use `DummyException` in tests, it can be compared:

```scala
val dummy = DummyException("Boom!")

val f = {
 ...
 Task.raiseError(dummy)
}

s.tick()

assertEquals(f.value, Some(Failure(dummy))
```

Otherwise looks good 👍 
```

Reply:
oh I see! cool will change that